From 860b2b3b1d616d357bab66c794654315b307b5db Mon Sep 17 00:00:00 2001
From: Michael Li <michael.li@netgear.com>
Date: Wed, 12 Jun 2019 14:17:56 +0800
Subject: [PATCH] RNAS-9045: Revert patch "s3-smbd: use fruit:model string for
 mDNS registration".

Looks like OSX doesn't have icon MacSamba and I tried to set the icon, it doesn't
show the old display icon, so revert this to get the old icon show.
---
 source3/smbd/avahi_register.c | 27 ---------------------------
 1 file changed, 27 deletions(-)

diff --git a/source3/smbd/avahi_register.c b/source3/smbd/avahi_register.c
index 99c16e7..50462b5 100644
--- a/source3/smbd/avahi_register.c
+++ b/source3/smbd/avahi_register.c
@@ -111,10 +111,8 @@ static void avahi_client_callback(AvahiClient *c, AvahiClientState status,
 		int dk = 0;
 		AvahiStringList *adisk = NULL;
 		AvahiStringList *adisk2 = NULL;
-		AvahiStringList *dinfo = NULL;
 		const char *hostname = NULL;
 		enum mdns_name_values mdns_name = lp_mdns_name();
-		const char *model = NULL;
 
 		DBG_DEBUG("AVAHI_CLIENT_S_RUNNING\n");
 
@@ -198,31 +196,6 @@ static void avahi_client_callback(AvahiClient *c, AvahiClientState status,
 			}
 		}
 
-		model = lp_parm_const_string(-1, "fruit", "model", "MacSamba");
-
-		dinfo = avahi_string_list_add_printf(NULL, "model=%s", model);
-		if (dinfo == NULL) {
-			DBG_DEBUG("avahi_string_list_add_printf"
-				  "failed: returned NULL\n");
-			avahi_entry_group_free(state->entry_group);
-			state->entry_group = NULL;
-			break;
-		}
-
-		error = avahi_entry_group_add_service_strlst(
-			    state->entry_group, AVAHI_IF_UNSPEC,
-			    AVAHI_PROTO_UNSPEC, 0, hostname,
-			    "_device-info._tcp", NULL, NULL, 0,
-			    dinfo);
-		avahi_string_list_free(dinfo);
-		if (error != AVAHI_OK) {
-			DBG_DEBUG("avahi_entry_group_add_service failed: %s\n",
-				  avahi_strerror(error));
-			avahi_entry_group_free(state->entry_group);
-			state->entry_group = NULL;
-			break;
-		}
-
 		error = avahi_entry_group_commit(state->entry_group);
 		if (error != AVAHI_OK) {
 			DBG_DEBUG("avahi_entry_group_commit failed: %s\n",
-- 
2.1.4

